statick_tool/plugins/reporting/code_climate_reporting_plugin.py

```diff
@@ -0,0 +1,143 @@
+"""Prints the Statick reports out to the terminal or file in JSON format."""
+import hashlib
+import json
+import logging
+import os
+from collections import OrderedDict
+from typing import Any, Dict, List, Optional, Tuple
+
+from statick_tool.issue import Issue
+from statick_tool.package import Package
+from statick_tool.reporting_plugin import ReportingPlugin
+
+
+class CodeClimateReportingPlugin(ReportingPlugin):
+    """Prints the Statick reports out to the terminal or file in JSON format."""
+
+    def get_name(self) -> str:
+        """Return the plugin name."""
+        return "code_climate"
+
+    def report(
+        self, package: Package, issues: Dict[str, List[Issue]], level: str
+    ) -> Tuple[Optional[None], bool]:
+        """Go through the issues list and print them in JSON format.
+
+        Args:
+            package (:obj:`Package`): The Package object that was analyzed.
+            issues (:obj:`dict` of :obj:`str` to :obj:`Issue`): The issues
+                found by the Statick analysis, keyed by the tool that found
+                them.
+            level: (:obj:`str`): Name of the level used in the scan.
+        """
+        if not self.plugin_context or not self.plugin_context.config:
+            return None, False
+
+        file_output = self.plugin_context.config.str_to_bool(
+            self.plugin_context.config.get_reporting_config(
+                self.get_name(), level, "files"
+            )
+        )
+        terminal_output = self.plugin_context.config.str_to_bool(
+            self.plugin_context.config.get_reporting_config(
+                self.get_name(), level, "terminal"
+            )
+        )
+        gitlab = self.plugin_context.config.str_to_bool(
+            self.plugin_context.config.get_reporting_config(
+                self.get_name(), level, "gitlab"
```

Comment:
This might be better called `gitlab_only` but I wanted to keep it short.
It's explained down on line 111